### ISSUE TITLE
Latest version (0.275) not launching dev container

### ISSUE THREAD

**Issue by BIGduzy** [microsoft/vscode-remote-release#7978]
This morning I could not reopen in the dev container. Rebuilding (with and without cache) also did not work. It turned out the plugin had automatically updated and after going back to version 0.266.1 everything worked as expected. After the container was running I could update back to 0.275 and it seemed to work.

- Version: 1.75.0 (user setup)
- Commit: e2816fe719a4026ffa1ee0189dc89bdfdbafb164
- Date: 2023-02-01T15:23:45.584Z
- Electron: 19.1.9
- Chromium: 102.0.5005.194
- Node.js: 16.14.2
- V8: 10.2.154.23-electron.0
- OS: Windows_NT x64 10.0.22621
- Sandboxed: No
- Remote Extension/Connection Type: Containers/WSL2

Steps to Reproduce:

1. Install version 0.275
2. Dev Containers: Reopen in Container (have no container running)
3. Nothing happens
4. Error printed in the Extension Host output

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

Logs:
```
2023-02-06 10:46:54.744 [error] Error: Illegal argument: base
	at v (c:\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:7:1267)
	at new ei (c:\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:94:44917)
	at Wne (c:\Users\Nok\.vscode\extensions\ms-vscode-remote.remote-containers-0.275.0\dist\extension\extension.js:337:23068)
	at hE (c:\Users\Nok\.vscode\extensions\ms-vscode-remote.remote-containers-0.275.0\dist\extension\extension.js:337:20963)
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async f (c:\Users\Nok\.vscode\extensions\ms-vscode-remote.remote-containers-0.275.0\dist\extension\extension.js:337:9132)
	at async s.h (c:\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:96:108008) remote-containers.reopenInContainer {"value":"ms-vscode-remote.remote-containers","c":"ms-vscode-remote.remote-containers"}
```

**Comment by chrmarti** [comment]
Was the window opened on a single folder when this happened? Was it WSL folder?

**Comment by rjra100** [comment]
We're seeing the same issue with "Open Folder in Container". In our case, it is a WSL folder (path like `\\wsl$\Ubuntu-20.04\home\<me>\<workspace>`)
"Open Folder in Container" or "Open Folder" followed by "Reopen in Container" pause for a moment, then produce "Illegal Argument: base". Opening the same folder from the Recents list works fine.

The issue appears to have started with the 0.275.0 release; I've repro'd with 0.275.0, 0.275.1, and (pre-release) 0.277

**Comment by FaBjE** [comment]
I'm experiencing the same issue.
Info from my ticket that might help:

My code is located in my WSL filesystem for performance reasons.
Accessing it from the \wsl$\Ubuntu\home\ or \wsl.localhost\Ubuntu\home path makes no difference.

Pre-release version v0.277.0 has the same behavior
Rolling back to previous versions I was able to trace that the fault is introduced in v0.271.0
v0.270.0 - OK
v0.271.0 - Fails

If I open a container workspace from the "Recent" items list, the container is opened and working.

**Comment by BIGduzy** [comment]
> Was the window opened on a single folder when this happened? Was it WSL folder?

Yes, the project is directly cloned in WSL2 and opened from there. Tested version 2.75.1 this morning also which did not work also.

**Comment by chrmarti** [comment]
This is triggered when using the UNC path (any of the `\\wsl...` variants). The workaround is to install the WSL extension (https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl), first reopen the folder in WSL (`F1` > `WSL: Reopen Folder in WSL`) and then reopen the folder in a dev container.

**Comment by FaBjE** [comment]
@chrmarti 
Thank you. This solves my/the issue!

I would not call it the most user-friendly change, but it makes sense in a way.

**Comment by BIGduzy** [comment]
@chrmarti The workaround works for me as well. For now however I'll just use the 0.266.1 version since the WSL extension adds a few extra seconds and actions to the startup.

Thanks for the help.

**Comment by vpommier** [comment]
This behavior was not there way before, WSL path could be opened without doing this workaround.
To be honest it not convenient at all for the user, it is a lost of time.

**Comment by lbergman** [comment]
Not sure I understand the status of this issue. A cumbersome workaround was presented, and then it was closed as completed, but cannot see any reference to a fix.
Is the workaround considered a fix @chrmarti?

**Comment by bjtho08** [comment]
> The workaround is to install the WSL extension

Like @lbergman, I would like to know if the workaround is supposed to be a permanent fix? The issue was not present in version 0.266.1, so I would say this is a clear-cut case of a bug that needs a code fix, not a workaround consisting of extra steps needed by the end user.

**Comment by rjra100** [comment]
Perhaps @chrmarti could confirm, but I think there is a "real" fix pending; it's just not released yet. If you switch to the pre-release version of the extension (currently 0.279.0), the problem seems to be gone there.
